src/saveload/industry_sl.cpp

```diff
@@ -229,7 +228,7 @@ struct INDYChunkHandler : ChunkHandler {
 			} else if (IsSavegameVersionBefore(SLV_INDUSTRY_CARGO_REORGANISE)) {
 				LoadMoveAcceptsProduced(i, INDUSTRY_NUM_INPUTS, INDUSTRY_NUM_OUTPUTS);
 			}
-			Industry::IncIndustryTypeCount(i->type);
+			Industry::industries[i->type].push_back(i->index); // Assume savegame indices are sorted.
```

Comment:
What do they say about assuming? 😛 

Jokes aside, is this a safe assumption here? Honest question, not a critique.

Reply:
Yeah it's safe here, when saving the pool is iterated, and it's ordered.

Reply:
It won't happen with any savegame made by OpenTTD, but the savegame format itself could have out-of-order indices if a file as somehow modified.